test/foundry/GaugeController.sol

```diff
@@ -0,0 +1,220 @@
+
+        // Deploy new root gauge from rootGaugeFactory
+        vm.startPrank(DEPLOYER);
+        // _chain_id: uint256, _salt: bytes32, _name: String[32]
+        vm.recordLogs();
+        IRootGaugeFactoryLike(rootGaugeFactory).deploy_gauge(
+            42161,
+            keccak256(bytes("fUSDC-USDC pool V2")),
+            "fUSDC-USDC pool V2"
+        );
+        vm.stopPrank();
+        // test adding new gauge, voting for it, killing old one check for inflation params after kill, after checkpoint
+    }
+
+    function test_V2GaugeAdd() public {
+        // It should be possible to add a new gauge to the gauge controller that shares an lp with another added gauge
+        // Get the address of the newly deployed gauge
+        rootGaugefUSDCV2 = IRootGaugeFactoryLike(rootGaugeFactory).get_gauge(
+            42161,
+            8
+        );
+        console2.log("rootGaugefUSDCV2 deployed at: ", rootGaugefUSDCV2);
+
+        // Pretend to be the gauge controller owner and add the new gauge
+        vm.prank(IGaugeControllerLike(gaugeController).admin());
+        IGaugeControllerLike(gaugeController).add_gauge(rootGaugefUSDCV2, 0, 0);
+
+        // Prank as a user and vote for the new gauge
+        vm.startPrank(gaugeVoter);
+        uint256 weight = IGaugeControllerLike(gaugeController)
+            .gauge_relative_weight(rootGaugefUSDCV2);
+        uint256 votedUserSlope = IGaugeControllerLike(gaugeController)
+            .vote_user_slopes(gaugeVoter, rootGaugefUSDCV2)
+            .slope;
+        uint256 inflationRateBefore = IRootGaugeLike(rootGaugefUSDCV2)
+            .inflation_params()
+            .rate;
+        // remove old votes for user
+        IGaugeControllerLike(gaugeController).vote_for_gauge_weights(
+            fraxGauge,
+            0
+        );
+        // vote for new gauge
+        IGaugeControllerLike(gaugeController).vote_for_gauge_weights(
+            rootGaugefUSDCV2,
+            10000
+        );
+        vm.stopPrank();
+        vm.warp(block.timestamp + 604800);
+        IGaugeControllerLike(gaugeController).checkpoint();
+        uint256 weightAfter = IGaugeControllerLike(gaugeController)
+            .gauge_relative_weight(rootGaugefUSDCV2);
+        uint256 votedUserSlopeAfter = IGaugeControllerLike(gaugeController)
+            .vote_user_slopes(gaugeVoter, rootGaugefUSDCV2)
+            .slope;
+        uint256 inflationRateAfter = IRootGaugeLike(rootGaugefUSDCV2)
+            .inflation_params()
+            .rate;
+        console2.log("votedUserSlope: ", votedUserSlope);
+        console2.log("votedUserSlopeAfter: ", votedUserSlopeAfter);
+        console2.log("weight: ", weight);
+        console2.log("weightAfter: ", weightAfter);
+        console2.log("inflationRateBefore: ", inflationRateBefore);
+        console2.log("inflationRateAfter: ", inflationRateAfter);
+        assert(weightAfter > weight);
+        assert(
+            votedUserSlopeAfter > votedUserSlope
+        );
```

Comment:
Since the test is going to behave deterministrically, we can also used fixed numbers for asserts